.github/workflows/build_deploy_exm_fe.yml

```diff
@@ -0,0 +1,64 @@
+name: Build and Deploy EXM FE
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Node.js
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
+
+    - name: Install dependencies
+      run: npm install
+      working-directory: expenses-manager-frontend/src/main/frontend
+
+    - name: Build Angular application
+      run: npm run build --prod
+      working-directory: expenses-manager-frontend/src/main/frontend
+
+    - name: Azure CLI Login
+      uses: azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+    - name: Get VM Public IP
+      id: get-vm-ip
+      run: |
+        az vm list-ip-addresses --name ${{ secrets.EXM_VM_NAME }} --resource-group ${{ secrets.RESOURCE_GROUP_NAME }} --query "[0].virtualMachine.network.publicIpAddresses[0].ipAddress" -o tsv
+      continue-on-error: true
+
+    - name: SSH into VM and install Nginx
+      run: |
+        ssh -o StrictHostKeyChecking=no $admin_username@${{ steps.get-vm-ip.outputs.vm_ip }} << 'EOF'
+          sudo apt-get update
+          sudo apt-get install nginx -y 
+        EOF
+      env:
+        admin_username: ${{ secrets.ADMIN_USERNAME }}
+
+    - name: Copy build files to VM
+      run: |
+        scp -o StrictHostKeyChecking=no -r dist/* $admin_username@${{ steps.get-vm-ip.outputs.vm_ip }}:/var/www/html/
```

Comment:
`dist/*` nu o sa functioneze pentru ca nu exista. poti seta CWD in `expenses-manager-frontend/src/main/frontend` sau sa inlocuiesti cu `expenses-manager-frontend/src/main/frontend/dist` in comanda.